.github/actions/verify-badge/action.yaml

```diff
@@ -0,0 +1,257 @@
+
+          # Should consider security options during implementation to provide the most secure default possible (i.e. SAML w/SCIM vs OIDC).
+          SSO_CLIENT_PROTOCOL=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.sso[].protocol')
+          if [ "$SSO_CLIENT_PROTOCOL" != "saml" ]; then
+            echo "❌ SAML is not the default protocol"
+            exit 1
+          fi
+          echo "✅ Default protocol is SAML"
+
+          # Should name the client <App> Login (i.e. Mattermost Login) to provide login UX consistency.
+          SSO_CLIENT_NAME=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.sso[].name')
+          if [ "$SSO_CLIENT_NAME" != "${UDS_PACKAGE_NAME^} Login" ]; then
+            echo "❌ SSO client name not in correct format"
+            exit 1
+          fi
+          echo "✅ SSO client name is correct"
+
+          # Should clearly mark the client id with the group and app name uds-<group>-<application> (i.e. uds-swf-mattermost) to provide consistency in the Keycloak UI.
+          SSO_CLIENT_ID=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.sso[].clientId')
+          if [ "$SSO_CLIENT_ID" != "uds-package-${UDS_PACKAGE_NAME}-${SSO_CLIENT_PROTOCOL}" ]; then
+            echo "❌ SSO client id not in the format uds-<group>-<application>-protocol"
+            exit 1
+          fi
+          echo "✅ SSO client id is correct"
+
+          ## May end any generated secrets with -sso to easily locate them when querying the cluster.
+          SSO_SECRET_NAME=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.sso[].secretName')
+          if [ "$SSO_SECRET_NAME" != "" ]; then
+            echo "ℹ️ SSO secret name is defined"
+            if [[ "$SSO_SECRET_NAME" != *-sso ]]; then
+              echo "❌ SSO secret name not in the format <application>-sso"
+              exit 1
+            fi
+            echo "✅ SSO secret name is in the correct format"
+          fi
+        else
+          echo "::warning ::⚠️ No SSO configuration found, review needed"
+        fi
+
+        echo
+
+        # Prometheus
+
+        ## Must implement monitors for each application metrics endpoint using it's built-in chart monitors, the Package CR monitor key, or manual monitors in the config chart.
+
+        PKG_MONITOR_COUNT=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.monitor | length')
+
+        if [ "$PKG_MONITOR_COUNT" -eq 0 ]; then
+          echo "ℹ️ No monitors defined in the package, checking for ServiceMonitors"
+
+          # Check for built-in serviceMonitors
+          SERVICE_MONITOR_COUNT=$(kubectl get ServiceMonitor -n "${NAMESPACE}" -o json | jq -r '.items | length')
+          if [ "$SERVICE_MONITOR_COUNT" -eq 0 ]; then
+            echo "ℹ️ No ServiceMonitors defined, checking for PodMonitors"
+
+            # Check for built-in podMonitors
+            POD_MONITOR_COUNT=$(kubectl get PodMonitor -n "${NAMESPACE}" -o json | jq -r '.items | length')
+            if [ "$POD_MONITOR_COUNT" -eq 0 ]; then
```

Comment:
So a couple things here. This actually is MUST (if possible)... For example we dont have monitoring on mattermost by default, despite a flag to enable it existing in the package because its locked behind a license. So its there, but would show up as not on this check.

Additionally, this doesnt actually check the intention of the rule, which is that ALL possible monitors are implemented. For example, gitlab has several. This would pass if only one was implemented, which is potentially misleading.

Reply:
Gave extra context in the messages on success